src/reference/zilla.yaml/binding-grpc-kafka.md

```diff
@@ -0,0 +1,414 @@
+  routes:
+    - when:
+        - method: example.FanoutService/*
+          metadata:
+            custom-text: custom value
+            custom-binary:
+              base64: Y3VzdG9tIHZhbHVl
+      exit: kafka_cache_client0
+      with:
+        capability: fetch
+        topic: messages
+        filters:
+          key: custom-key
+          headers:
+            custom-text: custom-value
+    - when:
+        - method: example.EchoService/*
+          metadata:
+            custom-text: custom value
+            custom-binary:
+              base64: Y3VzdG9tIHZhbHVl
+      exit: kafka_cache_client0
+      with:
+        capability: produce
+        topic: requests
+        acks: leader_only
+        key: custom-key
+        overrides:
+          custom-text: custom-value
+        reply-to: responses
+```
+
+## Summary
+
+The `proxy` kind `grpc-kafka` binding adapts `grpc` request-response streams to `kafka` topic streams.
+
+## Fetch capability
+
+Routes with `fetch` capability map `grpc` `Empty` requests to a `kafka` topic, supporting filtered retrieval of messages with a specific key or headers, or unfiltered retrieval of all messages in the topic merged into a unified response.
+
+Filtering can be performed by `kafka` message key, message headers, or a combination of both message key and headers.
+
+Reliable message delivery is achieved by capturing value of the `reliability` `field` injected into each response stream message at the `grpc` client, and replaying the value via the `reliability` `metadata` header when reestablishing the stream with a new `grpc` request.
+
+## Produce capability
+
+Routes with `produce` capability map any `grpc` request-response to a correlated stream of `kafka` messages. The `grpc` request message(s) are sent to a `requests` topic, with a `zilla:correlation-id` header. When the request message(s) are received and processed by the `kafka` `requests` topic consumer, it produces response message(s) to the `responses` topic, with the same `zilla:correlation-id` header to correlate the response.
+
+Requests including an `idempotency-key` `grpc` metadata header can be replayed and safely receive the same response. This requires the `kafka` consumer to detect and ignore the duplicate request with the same `idempotency-key` and `zilla:correlation-id`.
+
+## Configuration
+
+:::: note Properties
+
+- [kind\*](#kind)
+- [options](#options)
+  - [options.idempotency](#options-idempotency)
+    - [idempotency.metadata](#idempotency-metadata)
+  - [options.correlation](#options-correlation)
+    - [correlation.headers](#correlation-headers)
+    - [headers.service](#headers-service)
+    - [headers.method](#headers-method)
+    - [headers.correlation-id](#headers-correlation-id)
+    - [headers.reply-to](#headers-reply-to)
+- [routes](#routes)
+- [routes\[\].guarded](#routes-guarded)
+- [routes\[\].when](#routes-when)
+  - [when\[\].method](#when-method)
+  - [when\[\].metadata](#when-metadata)
```

Comment:
```suggestion
  - [when\[\].metadata](#when-metadata)
      - [metadata.base64](#metadata-base64)
```

This was omitted in both places. If that is intentional, we could mention the `base64` capability a different way vs have the inconsistency?